### DISCUSSION TITLE
[Feature] Persist file that yet doestn't exist

### DISCUSSION THREAD

**Discussion by zStruCat** [ScoopInstaller/Scoop#4758] [Help and Questions] [answered]
## Motivation
I was trying to make a manifest for Libgen-Desktop, a program that will create a database named "libgen.db" to store data under its folder after first run. However, "libgen.db" in manifest.persist resulting in creating a folder immediately after installation using scoop. Perhaps there is a way to workaround it, which i don't know.

## Feature Request
Is there a way a way to solve this? Perhaps scoop could
A.  Move and create softlink or junction of the file not immediately, but defered to some moment (perhaps invoking reset?) when the files or folders have been created by the software. or
B. Just instruct scoop to copy the files around when updating in the manifest.

Also, perhaps #3212 could be useful for this?

**Comment by rashil2000** [comment, marked as answer]
Scoop is, for the most part, stateless, so option A would not work.

You can either:
- create an empty file in pre_install
- do a manual persist (option B)
